train_segmentation_model.py

```diff
@@ -67,6 +67,16 @@ def train(gcp_bucket, config_file, random_module_global_seed, numpy_random_globa
     logs_dir = Path(model_dir, 'logs')
     logs_dir.mkdir(parents=True)
 
+    path_pre_trained_model = None
+    pre_trained_model_config = None
+    if pre_trained_model_id is not None:
```

Comment:
@Josh-Joseph, we may want to use `assert`s to enforce continuity of pretrained model config/metadata and current model config, e.g., make sure number of classes matches, same model architecture, same loss function, same color channels, same crop target size, etc. thoughts? some of these inconsistencies could cause train failure and would be less obvious at that point

Reply:
any thoughts, @Josh-Joseph ?

Reply:
yeah I think this is a great idea. @rak5216 I think you covered it -- generally I'd include asserts for:
- pretrained model name (is helpful to know if you, for example, misspelled something)
- input sizes
- architectures and pretrained backbones that we know don't play nicely together (you can imagine 6 months from now us completely forgetting which don't work and it'd be nice for an assert to tell us that)

Reply:
I created the asserts: 
![image](https://user-images.githubusercontent.com/43469359/97361993-c083c200-1876-11eb-8616-9c3b7a2da019.png)


E.g if the backbone of the trained and pretrained models are different, it fails (added the prints to make sure it was doing what I expected,  going to remove them now): 
![image](https://user-images.githubusercontent.com/43469359/97361834-861a2500-1876-11eb-9e0b-899c222256c9.png)

if everything is equal, it works: 
![image](https://user-images.githubusercontent.com/43469359/97361894-9af6b880-1876-11eb-8b6a-717f9d77e98a.png)

Anyhing else missing @rak5216 @Josh-Joseph ?